[GC] Added documentation for GC on Azure (#3790)
treeverse/lakeFS: docs/reference/garbage-collection.md

  -c spark.hadoop.fs.s3a.secret.key=<S3_SECRET_KEY> \
  <APPLICATION-JAR-PATH> \
  example-repo us-east-1
  ```
  </div>

  <div markdown="1" id="azure-option">

> The relevant Spark client for running GC on Azure is `lakefs-spark-client-312-hadoop3` only. This client is compiled for Spark 3.1.2 with Hadoop 3.2.1, but can work for other Spark versions and higher Hadoop versions.

Reviewer: ```suggestion
To run GC on Azure, use `lakefs-spark-client-312-hadoop3` only. This client is compiled for Spark 3.1.2 with Hadoop 3.2.1, but may work with other Spark versions and higher Hadoop versions. Specifically, this client was tested on Databricks runtime DBR 11.0 (Spark 3.3.0, 3.3.2).  

```

You should run the following command to make the garbage collector start running:

  ```bash
spark-submit --class io.treeverse.clients.GarbageCollector \
  -c spark.hadoop.lakefs.api.url=https://lakefs.example.com:8000/api/v1  \
  -c spark.hadoop.lakefs.api.access_key=<LAKEFS_ACCESS_KEY> \
  -c spark.hadoop.lakefs.api.secret_key=<LAKEFS_SECRET_KEY> \
  -c spark.hadoop.fs.azure.account.key.<AZURE_STORAGE_ACCOUNT>.dfs.core.windows.net=<AZURE_STORAGE_ACCESS_KEY> \

>   s3://treeverse-clients-us-east/lakefs-spark-client-312-hadoop3/0.1.8/lakefs-spark-client-312-hadoop3-assembly-0.1.8.jar \

Reviewer: We will need to update the metadata client version after publishing a new one next week

  example-repo
  ```

> In case you don't have `hadoop-azure` package as park of your environment, you should add the package to your spark-submit with `--packages org.apache.hadoop:hadoop-azure:3.2.1`

Reviewer: ```suggestion
In case you don't have `hadoop-azure` package as part of your environment, you should add the package to your spark-submit with `--packages org.apache.hadoop:hadoop-azure:3.2.1`
```

> To GC to work on Azure blob, soft delete should be disabled.

Reviewer: ```suggestion
Note: For GC to work on Azure blob, soft delete should be disabled.
```

Reviewer: Can you please link to some Azure docs that explain how to disable soft delete?

Reviewer 2: ```suggestion
Note: Disable soft delete on the blob.  Otherwise GC cannot ...
```

- Uses the [active voice](https://docs.google.com/document/d/1vif05nDiKZYPwYdINLoq4BJ6j7bBIQ6ZA2dt7McuzyY/edit#)
- Starts with the call to action.
- (If we can, we should also explain why, in a second sentence.)
